src/core/dns_cache.c

```diff
@@ -3292,6 +3292,9 @@ inline static int dns_naptr_sip_resolve(struct dns_srv_handle* h,  str* name,
 		}
 		try_lookup_naptr = 1;
 	}
+	/* check if it's an ip address, dns_srv_sip_resolve will return the right failure */
```

Comment:
@jchavanton, can you give an example for your case?
I suppose that your check will never work, because we have the same check in line 3279 for first call of dns_naptr_sip_resolve

Reply:
Hi, thanks for having a look.

Since you moved `dns_get_entry` outside of the following check, it can now be executed without checking for IP.


```
if (((h->srv==0) && (h->a==0)) && /* first call */
			 proto && port && (*proto==0) && (*port==0)){
```